Challo/Challo/Modules/Trails/TrailBooking/TrailBookingInteractor.swift

```diff
@@ -14,19 +14,40 @@ class TrailBookingInteractor: InteractorProtocol {
 
     private let trailRepository: TrailRepositoryProtocol
     private let guideRepository: GuideRepositoryProtocol
+    private let bookingRepository: BookingRepositoryProtocol
     private let bookingAPI: BookingAPIProtocol
     private let userState: UserStateProtocol
+    private var originalGuides = [Guide]()
+    private var datesWithExistingBookings = Set<Date>()
```

Comment:
Also have to keep track of the guide for existing bookings, since you can book a trail with a guide who doesnt have a booking on that date.

Probably can track it via a tuple

Reply:
Hmm I'm not sure what you mean. The `datesWithExistingBookings` is just dates where the user already have a booking, so he/she shouldnt be able to book any other trails on that day right?